openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java

```diff
@@ -0,0 +1,459 @@
+package org.heigit.ors.fastisochrones;
+
+import com.carrotsearch.hppc.IntHashSet;
+import com.carrotsearch.hppc.IntObjectHashMap;
+import com.carrotsearch.hppc.IntObjectMap;
+import com.carrotsearch.hppc.IntSet;
+import com.carrotsearch.hppc.cursors.IntCursor;
+import com.carrotsearch.hppc.cursors.IntObjectCursor;
+import com.graphhopper.routing.util.EdgeFilter;
+import com.graphhopper.storage.GraphHopperStorage;
+import com.graphhopper.storage.NodeAccess;
+import com.graphhopper.util.EdgeExplorer;
+import com.graphhopper.util.EdgeIterator;
+import com.graphhopper.util.PointList;
+import com.vividsolutions.jts.geom.*;
+import com.vividsolutions.jts.index.quadtree.Quadtree;
+import org.heigit.ors.fastisochrones.partitioning.storage.CellStorage;
+import org.heigit.ors.fastisochrones.partitioning.storage.IsochroneNodeStorage;
+import org.heigit.ors.isochrones.builders.concaveballs.PointItemVisitor;
+import org.opensphere.geometry.algorithm.ConcaveHull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.heigit.ors.fastisochrones.partitioning.FastIsochroneParameters.getMaxCellNodesNumber;
+import static org.heigit.ors.fastisochrones.partitioning.FastIsochroneParameters.isSupercellsEnabled;
+
+/**
+ * Calculates Outlines (Contour) of cells.
+ * Contours are concave hulls of a given set of points.
+ * Additionally, contours for supercells can be created.
+ * <p>
+ *
+ * @author Hendrik Leuschner
+ */
+public class Contour {
+    private static final int MIN_EDGE_LENGTH = 400;
+    private static final int MAX_EDGE_LENGTH = Integer.MAX_VALUE;
+    private static final int SUPER_CELL_HIERARCHY_LEVEL = 3;
```

Comment:
I do not think that its necessary or useful for the user to have this exposed in the app.config. I think most people who would like to use fast isochrones are already confused enough by the maxcellnodes number. The supercells are in principle only part of a black box and to use the code, you dont even need to know they exist.
There are a lot more factors that could be used for benchmarking and tuned for performance. But I don't want to expose any of them to the app.config.

Reply:
It's your call.

Reply:
I'll leave it like it is then. Thinking about removing the cellNodes parameter from the app.config, though I think it might be useful to change this depending on the profile.